src/stores/room-list/ListLayout.ts

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2020 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+const TILE_HEIGHT_PX = 34;
+
+interface ISerializedListLayout {
+    numTiles: number;
+}
+
+export class ListLayout {
+    private _n = 0;
+
+    constructor(public readonly tagId) {
```

Comment:
this syntax is blowing my mind, also type?

Reply:
yea, it saves the `this.whatever = whatever` lines. I love it :)

Will add types.

Reply:
I wish I knew about it earlier, its game changing

Reply:
Pretty sure you tried to teach me the way earlier but it didn't apply as it was a destructured object param